### ISSUE TITLE
Add complex number Descriptor

### ISSUE THREAD

**Issue by Euler-37** [j3-fortran/fortran_proposals#212]
add descriptor to dispay format complex data
(1) `format(CF10.4)` and `format("(",F10.4,",",F10.4,")")`
(2) `format(CES12.5)` and `format("(",ES12.5,",",ES12.5,")")`
(3) Experimental , add python like format
    `format(ZF10.4)` and `format(F10.4,"+",F10.4,"j")`

**Comment by certik** [comment]
I think both of these could be implemented in [stdlib](https://github.com/fortran-lang/stdlib/) first.

**Comment by Euler-37** [comment]
> I think both of these could be implemented in [stdlib](https://github.com/fortran-lang/stdlib/) first.

Yes , [https://github.com/fortran-lang/stdlib/pull/445](url) they prepare to add function `disp` ,It is a good feature.

**Comment by klausler** [comment]
Examples (1) and (2) are surprising omissions in Fortran; implementing them would be straightforward and valuable.  List-directed and NAMELIST output of COMPLEX already exist in the language in this parenthesized form, with an adjustable comma character too.

Example (3) is more dubious.  The FORMAT edit descriptor "Z" is already taken for another purpose, and emitting a form of complex output that differs from those used in Fortran source code is less compelling.
